### PR TITLE
Digitizer for FDD

### PR DESCRIPTION
**Pull request by mbroz84**

### CODE REVIEW

**Comment by mbroz84** [comment]
What is the "BoostSerializedProcessing" whihc failed doing? How can I run it?

File: DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  ClassDefNV(ChannelData, 1);
+};
+
+using DigitBase = o2::dataformats::TimeStamp<double>;
+class Digit : public DigitBase
+{
+ public:
+  Digit() = default;
+
+  Digit(std::vector<ChannelData> channelData, Double_t time, uint16_t bc, uint32_t orbit, std::vector<Bool_t> triggers)
```
Comment (sawenzel) [inline DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h]: Do you really need to make a digit using existing outer containers? (Could the digit not manage it's own containers?)
I would in any case still prefer to have `const&` here.

File: Detectors/FIT/FDD/simulation/include/FDDSimulation/DigitizerTask.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#include "SimulationDataFormat/MCTruthContainer.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+#include "DataFormatsFDD/MCLabel.h"
+
+namespace o2
+{
+namespace fdd
+{
+
+class DigitizerTask : public FairTask
```
Comment (sawenzel) [inline Detectors/FIT/FDD/simulation/include/FDDSimulation/DigitizerTask.h]: we don't use FairTasks anymore

Comment (mbroz84) [reply Detectors/FIT/FDD/simulation/include/FDDSimulation/DigitizerTask.h]: Task removed.

File: DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h (hunk 1 above)
Comment (mbroz84) [reply DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h]: Its very handy like that... Where exactly do you want the const& , not clear to me?

Comment (sawenzel) [reply DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h]: `Digit(std::vector<ChannelData> const& channelData,...)`. Without the reference, the vector will be copied into a tmp container when you call the function (as a general rule always pass complicated objects by reference and not by value).

Comment (mbroz84) [reply DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h]: Ok, thanks for explanation. Done now.

**Comment by davidrohr** [comment]
@sawenzel @mbroz84 : We will apply the global CMake reformatting of O2 this weekend, I'd suggest to try to merge this beforehand.

File: DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  ClassDefNV(ChannelData, 1);
+};
+
+using DigitBase = o2::dataformats::TimeStamp<double>;
+class Digit : public DigitBase
+{
+ public:
+  Digit() = default;
+
+  Digit(std::vector<ChannelData> const& channelData, Double_t time, uint16_t bc, uint32_t orbit, std::vector<Bool_t> triggers)
```
Comment (sawenzel) [inline DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h]: can you also pass the `std::vector<bool>` by const reference?

**Comment by sawenzel** [comment]
Can be merged after another small fix on vector.

File: DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h (hunk 3 above)
Comment (mbroz84) [reply DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/Digit.h]: Done.

**Comment by davidrohr** [comment]
OK, looks good to me. @sawenzel : Should we squash and merge?

**Pull request merged**